HopsanGenerator/include/hopsangenerator.h

```diff
@@ -28,6 +28,12 @@ extern "C" {
     HOPSANGENERATOR_DLLAPI bool callCheckComponentLibrary(const char* libraryXMLPath, messagehandler_t messageHandler=0, void* pMessageObject=0);
 
     HOPSANGENERATOR_DLLAPI bool callExeExportGenerator(const char* outputPath, void* pHopsanSystem,  const char* const externalLibraries[], const int numLibraries, const char* hopsanInstallPath, const char* compilerPath, int architecture=64, messagehandler_t messageHandler=0, void* pMessageObject=0);
+
+    HOPSANGENERATOR_DLLAPI bool callAddComponentToLibrary(const char* libraryXMLPath, const char* typeName, const char* displayName, const char* cqsType, const char * const constantNames[], const int numConstantNames, const char * const constantDisplayNames[], const int numConstantDisplayNames, const char * const constantUnits[], const int numConstantUnits, const char * const constantInits[], const int numConstantInits, const char * const inputNames[], const int numInputNames, const char * const inputDescriptions[], const int numInputDescriptions, const char * const inputUnits[], const int numInputUnits, const char * const inputInits[], const int numInputInits, const char * const outputNames[], const int numOutputNames, const char * const outputDescriptions[], const int numOutputDescriptions, const char * const outputUnits[], const int numOutputUnits, const char * const outputInits[], const int numOutputInits, const char * const portNames[], const int numPortNames, const char * const portDescriptions[], const int numPortDescriptions, const char * const portTypes[], const int numPortTypes, const int portsRequired[], const int numPortsRequired, messagehandler_t messageHandler=0, void* pMessageObject=0);
```

Comment:
nice function :) actually we should invert this relationship so that the GUI uses the Qt api directly instead of going through the c-api. The C-api is only required when used from HopsanCLI. Then we could reduce the C-api to only include functions for those things needed by the CLI. I will do that at some point. (no change needed now)

Reply:
Sounds good to me!